v1/charm-dev.yaml

```diff
@@ -0,0 +1,77 @@
+description: A development and testing environment for charmers
+version: latest
+
+instances:
+  charm-dev:
+    image: 20.04
+    limits:
+      min-cpu: 2
+      min-mem: 4G
+      min-disk: 10G
+    timeout: 900
+    cloud-init:
+      vendor-data: |
+        snap:
+          commands:
+          - snap install --classic juju
+          - snap install --classic microk8s
+          - snap alias microk8s.kubectl kubectl
+          - snap install --classic charmcraft
+          - snap install yq
+
+        runcmd:
+        - |
+          # disable swap
+          sysctl vm.swappiness=0
+          echo "vm.swappiness = 0" | tee -a /etc/sysctl.conf
+
+        - |
+          # disable unnecessary services
+          systemctl disable man-db.timer man-db.service --now
+          systemctl disable apport.service apport-autoreport.service  --now
+          systemctl disable apt-daily.service apt-daily.timer --now
+          systemctl disable apt-daily-upgrade.service apt-daily-upgrade.timer --now
+          systemctl disable unattended-upgrades.service --now
+          systemctl disable motd-news.service motd-news.timer --now
+          systemctl disable bluetooth.target --now
+          systemctl disable ua-messaging.service ua-messaging.timer --now
```

Comment:
Disable cups as well if it's in there. :)

Reply:
Eh. If only minimal images had the generic kernel, we could start there…

Reply:
cups isn't there :)